network/src/overlay/background_tasks.rs

```diff
@@ -166,6 +180,9 @@ impl OverlayServiceInner {
                                 }
                             },
                         );
+                        sign.rebuild(iter, |_| {
+                            tokio::time::interval(this.config.public_overlay_peer_sign_period)
+                        });
```

Comment:
```rs
sign.rebuild(iter, |_| {
    tokio::time::interval_at(
        Instant::now() + this.config.public_overlay_peer_sign_period,
        this.config.public_overlay_peer_sign_period,
    )
});
```

Otherwise, you are waiting `public_overlay_peer_sign_period` before the first sign